tests/e2e/kserve_test.go

```diff
@@ -416,3 +423,144 @@ func (tc *KserveTestCtx) validateTemplatedResourceOwnerRefsAndLabels(expectOwned
+		)),
+	)
+
+	t.Log("Verifying Kserve is ready")
+	tc.verifyKserveReady(t, metav1.ConditionTrue)
+
+	t.Log("Verifying custom secret is created")
+	_, err := cluster.GetSecret(tc.g.Context(), tc.g.Client(), originalDSCI.Spec.ServiceMesh.ControlPlane.Namespace, customSecretName)
+	tc.g.Expect(err).NotTo(HaveOccurred())
+
+	t.Log("Restoring original DSC state")
+	tc.restoreDSCSpec(t, originalDSC)
+
+	t.Log("Deleting custom secret")
+	tc.DeleteResource(
+		WithMinimalObject(
+			gvk.Secret,
+			types.NamespacedName{Namespace: originalDSCI.Spec.ServiceMesh.ControlPlane.Namespace, Name: customSecretName},
+		),
+	)
+
+	t.Log("Custom certificate creation test completed successfully")
+}
+
+// ValidateNoInvalidCustomCertificateCreation tests rejection of an invalid custom certificate for OpenshiftDefaultIngress.
+func (tc *KserveTestCtx) ValidateNoInvalidCustomCertificateCreation(t *testing.T) {
+	t.Helper()
+
+	originalDSC := tc.FetchDataScienceCluster()
+	originalDSCI := tc.FetchDSCInitialization()
+	invalidCustomSecretName := "&invalid-secret-name"
+
+	t.Log("Configuring Kserve with OpenshiftDefaultIngress and invalid secret")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(testf.TransformPipeline(
+			testf.Transform(`.spec.components.kserve.serving.managementState = "%s"`, operatorv1.Managed),
+			testf.Transform(`.spec.components.kserve.serving.ingressGateway.certificate.type = "%s"`, infrav1.OpenshiftDefaultIngress),
+			testf.Transform(`.spec.components.kserve.serving.ingressGateway.certificate.secretName = "%s"`, invalidCustomSecretName),
+		)),
+	)
+
+	t.Log("Verifying Kserve reports not ready due to invalid secret")
+	tc.verifyKserveReady(t, metav1.ConditionFalse)
+
+	t.Log("Verifying invalid secret is not created")
+	_, err := cluster.GetSecret(tc.g.Context(), tc.g.Client(), originalDSCI.Spec.ServiceMesh.ControlPlane.Namespace, invalidCustomSecretName)
+	tc.g.Expect(err).To(HaveOccurred())
+
+	tc.restoreDSCSpec(t, originalDSC)
+
+	t.Log("Invalid custom certificate creation test completed successfully")
+}
+
+func (tc *KserveTestCtx) verifyKserveReady(t *testing.T, expectedStatus metav1.ConditionStatus) {
+	t.Helper()
+	jqKserveReadyTrue := `
+	.status.conditions[]
+	| select(.type == "KserveReady")
+	| .status == "%s"`
+
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithCondition(
+			jq.Match(jqKserveReadyTrue, expectedStatus),
+		),
+		WithEventuallyTimeout(tc.TestTimeouts.mediumEventuallyTimeout),
+	)
+}
+
+func (tc *KserveTestCtx) restoreDSCSpec(t *testing.T, originalDSC *dscv1.DataScienceCluster) {
+	t.Helper()
+	t.Log("Restoring original DSC state")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(testf.Transform(`.spec = %s`, envtestutil.JSONMarshal(t, originalDSC.Spec))),
```

Comment:
To avoid the overhead of JSON serialization/deserialization, consider using a more direct approach. 
```suggestion
                 WithMutateFunc(testf.TransformSpecToUnstructured(originalDSC.Spec)),
```

You can also do:
```suggestion
                 WithMutateFunc(func(obj client.Object) error {
			dsc := obj.(*dscv1.DataScienceCluster)
			dsc.Spec = originalDSC.Spec
			return nil
		}),
```